tests/api/api/test_feature_sets.py

```diff
@@ -163,6 +272,92 @@ def test_delete_feature_set(db: Session, client: TestClient) -> None:
     _assert_list_objects(client, "feature_sets", project_name, None, count - 2)
 
 
+def test_add_multiple_times_same_uid(db: Session, client: TestClient) -> None:
+    project_name = f"prj-{uuid4().hex}"
+    name = "feature_set1"
+    feature_set = _generate_feature_set(name)
+
+    _assert_add_feature_set(client, project_name, feature_set, versioned=True)
+
+    response = client.post(
+        f"/api/projects/{project_name}/feature_sets?versioned=True", json=feature_set
+    )
+    assert response.status_code != HTTPStatus.OK.value
+
+    # Now test not-versioned add
+    name = "feature_set2"
+    feature_set = _generate_feature_set(name)
+    added_feature_set = _assert_add_feature_set(
+        client, project_name, feature_set, versioned=False
+    )
+    uid = added_feature_set["metadata"]["uid"]
+    assert uid.startswith("unversioned")
+
+    response = client.post(
+        f"/api/projects/{project_name}/feature_sets?versioned=False", json=feature_set
+    )
+    assert response.status_code != HTTPStatus.OK.value
+
+
+def test_multi_inserts_and_updates(db: Session, client: TestClient) -> None:
+    project_name = f"prj-{uuid4().hex}"
+    count = 5
+    for i in range(count):
+        name = f"fs_{i}"
+        feature_set = _generate_feature_set(name)
+        _assert_add_feature_set(client, project_name, feature_set)
```

Comment:
If this function create and assert, let's name it like that so will be easier to understand
`_create_feature_set_and_assert`